org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/jdt/JavaDebugServer.java

```diff
@@ -23,90 +23,118 @@
 import org.eclipse.jdt.ls.debug.internal.Logger;
 
 public class JavaDebugServer implements IDebugServer {
```

Comment:
Why not make the JavaDebugServer itself a singleton instead of hosting an internal class?